lib/crypto/sr25519/sr25519.go

```diff
@@ -48,6 +49,22 @@ type PrivateKey struct {
 	key *sr25519.SecretKey
 }
 
+// VerifySignature verifies a signature given a public key and a message
+func VerifySignature(publicKey, signature, message []byte) error {
+	pubKey, err := NewPublicKey(publicKey)
+	if err != nil {
+		return fmt.Errorf("sr25519: %w", err)
+	}
+
+	_, err = pubKey.Verify(message, signature)
```

Comment:
same here - please check `bool` as the signature might be invalid, but no error is returned

Reply:
its okay to do something like this:

```go
ok, err := pubKey.Verify(message, signature)
    if err != nil || !ok {
        return fmt.Errorf("ed25519: %w: for message 0x%x, signature 0x%x and public key 0x%x",
	    crypto.ErrSignatureVerificationFailed, message, signature, publicKey)
    }
```

since we don't return a bool anymore?